app/src/main/java/com/vero/tasky/agenda/data/repository/AgendaRepositoryImpl.kt

```diff
@@ -42,11 +36,12 @@ class AgendaRepositoryImpl(
         return merge(taskFlow, eventFlow, reminderFlow)
     }
 
-    override suspend fun updateAgendaForDay(timezone: String,
-                                            timestamp: Long): Result<Unit> {
+    override suspend fun updateAgendaForDay(timestamp: Long): Result<Unit> {
         return safeApiCall {
-            val networkAgendaItems = api.getAgendaForDay(timezone, timestamp)
-            saveAgendaItems(networkAgendaItems)
+            withContext(Dispatchers.IO) {
```

Comment:
Yes, I need to wrap not all of these methods, only saveAgendaItems (because Room will throw the Exception, that I'm saving info from the main thread)

Reply:
Room only throws that because you didn't use suspend functions there, so Room can't switch to the IO dispatcher under the hood (since `withContext` is only for suspend functions). Simply make the functions in your DAO all suspend functions and you don't need this `withContext` (only those that don't return a Flow)